### PR TITLE
Add rules S1751 and S3626 for TS/JS (S3626 new rule for JS)

### LINKED ISSUE #1494: Add rule S1751 (based on 'no-one-iteration-loop') for TS/JS

### LINKED ISSUE #1510: Add rule S3626 ('no-redundant-jump') for TS/JS (new for JS)

### PR DESCRIPTION
**Pull request by vilchik-elena**
Fixes #1510 and fixes #1494 

S3626:
For TS some issues requiring CFG are missing, still there are many new (probably our CFG impl was limited)

S1751:
For JS issue location changed (reported on unconditional jump before, now on the loop)

### CODE REVIEW

File: eslint-bridge/tests/rules/no-redundant-jump.test.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        }
+        if (b) {
+          console.log("b");
+          return;
+        }
+        console.log("useful");
+      }
+
+      while (x == 1) {
+        continue; // Ok, we ignore when 1 statement
```
Comment (quentin-jaquier-sonarsource) [inline eslint-bridge/tests/rules/no-redundant-jump.test.ts]: Why this exception? I did not find any documentation related to this.

File: eslint-bridge/tests/rules/no-redundant-jump.test.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
+import { RuleTester } from "eslint";
+
+const ruleTester = new RuleTester({ parserOptions: { ecmaVersion: 2018 } });
+import { rule } from "../../src/rules/no-redundant-jump";
+
+function invalid(code: string) {
```
Comment (quentin-jaquier-sonarsource) [inline eslint-bridge/tests/rules/no-redundant-jump.test.ts]: 💯

File: eslint-bridge/src/rules/no-redundant-jump.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+const message = "Remove this redundant jump.";
+const loops = "WhileStatement, ForStatement, DoWhileStatement, ForInStatement, ForOfStatement";
+
+export const rule: Rule.RuleModule = {
+  create(context: Rule.RuleContext) {
+    function reportIfLastStatement(node: estree.ContinueStatement | estree.ReturnStatement) {
+      const withArgument = node.type === "ContinueStatement" ? !!node.label : !!node.argument;
+      if (!withArgument) {
+        const ancestors = context.getAncestors();
+        const block = ancestors[ancestors.length - 1] as estree.BlockStatement;
```
Comment (christophe-zurn-sonarsource) [inline eslint-bridge/src/rules/no-redundant-jump.ts]: You could use `getParent` from "eslint-plugin-sonarjs/lib/utils/nodes" package here for simplicity

File: eslint-bridge/src/rules/no-redundant-jump.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    function reportIfLastStatementInsideIf(
+      node: estree.ContinueStatement | estree.ReturnStatement,
+    ) {
+      const ancestors = context.getAncestors();
+      const ifStatement = ancestors[ancestors.length - 2];
+      const upperBlock = ancestors[ancestors.length - 3] as estree.BlockStatement;
+      if (upperBlock.body[upperBlock.body.length - 1] === ifStatement) {
+        reportIfLastStatement(node as estree.ContinueStatement);
```
Comment (christophe-zurn-sonarsource) [inline eslint-bridge/src/rules/no-redundant-jump.ts]: the type assertion `node as estree.ContinueStatement` is not needed here, as `node` already has correct type

File: eslint-bridge/tests/rules/no-redundant-jump.test.ts (hunk 1 above)
Comment (vilchik-elena) [reply eslint-bridge/tests/rules/no-redundant-jump.test.ts]: Indeed I didn't document this exception, will add it to RSPEC now.
Idea is that between empty block and block containing single `continue`, second option is more readable.

File: eslint-bridge/src/rules/no-redundant-jump.ts (hunk 4 above)
Comment (vilchik-elena) [reply eslint-bridge/src/rules/no-redundant-jump.ts]: I wonder why SQ didn't report that..

**Pull request merged**